forte/processors/allennlp_processors.py

```diff
@@ -0,0 +1,145 @@
+}
+# pylint: enable=line-too-long
+
+
+class AllenNLPProcessor(PackProcessor):
+
+    # pylint: disable=attribute-defined-outside-init,unused-argument
+    def initialize(self, resource: Resources, configs: HParams):
+        self.processors = configs.processors
+        if self.processors is None or self.processors == "":
+            self.processors = self.default_configs()['processors']
+
+        if configs.output_format not in MODEL2URL:
+            raise ProcessorConfigError('Incorrect value for output_format')
+        model_url = MODEL2URL[configs.output_format]
+        self.predictor = Predictor.from_path(model_url)
+
+        self.overwrite_entries = configs.overwrite_entries
+        self.allow_parallel_entries = configs.allow_parallel_entries
+        if self.overwrite_entries:
+            logger.warning("`overwrite_entries` is set to True. Deleting an "
+                           "annotation doesn't guarantee deletion of all the "
+                           "associated links.")
+            if self.allow_parallel_entries:
+                logger.warning('Both `overwrite_entries` and '
+                               '`allow_parallel_entries` are True, all '
+                               'existing conflicting entries will be deleted.')
+        else:
+            if not self.allow_parallel_entries:
+                logger.warning('Both `overwrite_entries` and '
+                               '`allow_parallel_entries` are False, '
+                               'processor will only run if there are no '
+                               'existing conflicting entries.')
+
+    @staticmethod
+    def default_configs():
+        """
+        This defines a basic Hparams structure for AllenNLP.
+        :return:
+        """
+        return {
+            'processors': "tokenize,pos,depparse",
+            # format of the POS tags and dependency parsing
+            'output_format': "universal_dependencies",
+            # whether to overwrite the entries of the same Type as produced by
+            # this processor
+            'overwrite_entries': True,
+            # whether to allow similar new entries when they already exist, e.g.
+            # allowing new tokens with same spans, used only when
+            # `overwrite_entries` is False
+            'allow_parallel_entries': False
+        }
+
+    def _process(self, input_pack: DataPack):
+        # handle existing entries
+        self._process_existing_entries(input_pack)
+
+        for sentence in input_pack.get(Sentence):
+            result = self.predictor.predict(sentence=sentence.text)
+
+            if "tokenize" in self.processors:
+                # creating new tokens and dependencies
+                tokens = self._create_tokens(input_pack, sentence, result)
+                if "depparse" in self.processors:
+                    self._create_dependencies(input_pack, tokens, result)
+
+    def _process_existing_entries(self, input_pack):
```

Comment:
Looks like we can make a utility from this.

Reply:
Yes, this is part of what I want to do that in the refactoring of our tests.